custom profile field: Add tests for default external field reuse issue.

YJDave opened the pull request:
This commit adds test case for handling a case where realm
have non external account field with our default realm field
name.

Followup of #12913 

<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing Plan:** <!-- How have you tested? -->


**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

YJDave committed:
custom profile field: Modify field name duplication error message.

timabbott commented on zerver/views/custom_profile_fields.py:
```diff
@@ -108,8 +117,8 @@ def create_realm_custom_profile_field(request: HttpRequest,
-    except IntegrityError:
-        return json_error(_("A field with that label already exists."))
+        except IntegrityError:
+            return json_error(_("A field with that label already exists."))
```
I think it'd be cleaner to preserve the existing try/except structure and just do a second check of `if is_default_external_field`.  

That said, I think a better solution would be to keep the code as-is and just change "that label" to "label '%s'" passing the label value we were going to use; that would avoid adding extra strings and provide all the information the user needs to figure out they need to rename their field.

YJDave committed:
custom fields: Add test case for default external field name dupliation.

This commit adds test case for handling a case where realm
already have non external account field with our default realm
field name and try to add default field.

YJDave commented:
@timabbott updated!
@rishig do we want to add any other social media in default external accounts?

timabbott commented:
I think the word ordering is wrong; what I had in mind was "with label '{label}'" -- where we include the label in quotes in the string and put it after the word "label".

rishig commented:
I think just linkedin (along with the existing twitter and github).

timabbott commented:
@YJDave just a quick ping; I'd love to just get this merged as it's nearly done so we can fully advertise this feature in Zulip 2.1.
